lightkurve/seismology.py

```diff
@@ -0,0 +1,908 @@
+"""Defines the asteroseismology module"""
+from __future__ import division, print_function
+
+import copy
+import os
+import logging
+import warnings
+
+import numpy as np
+from matplotlib import pyplot as plt
+from mpl_toolkits.axes_grid1.inset_locator import inset_axes
+import astropy
+from astropy import units as u
+from astropy import constants as const
+from astropy.convolution import convolve, Box1DKernel, Gaussian1DKernel
+
+
+from uncertainties import ufloat
+from uncertainties import umath
+
+from . import MPLSTYLE
+
+from scipy.signal import find_peaks
+
+from .utils import LightkurveWarning
+
+
+log = logging.getLogger(__name__)
+
+#__all__ = ['estimate_radius','estimate_mass','estimate_logg']
+
+"""Global parameters for the sun"""
+NUMAX_SOL = ufloat(3090, 30) # microhertz | Huber et al. 2011
+DELTANU_SOL = ufloat(135.1, 0.1) # microhertz | Huber et al. 2011
+TEFF_SOL = ufloat(5772., 0.8) # Kelvin    | Prsa et al. 2016
+G_SOL = ((const.G * const.M_sun)/(const.R_sun)**2).to(u.cm/u.second**2) #cms^2
+
+class SeismologyButler(object):
+    '''Good day, I am the Seismology Butler, I am here to help clean up your seismic mess.
+    '''
+
+    def __init__(self, periodogram):
+        self.periodogram = periodogram
+        self.frequency = periodogram.frequency
+        self.power = periodogram.power
+        self._numax_result = None
+        self._deltanu_result = None
+
+#        super(LombScarglePeriodogram, self).__init__(*args, **kwargs)
+
+    def __repr__(self):
+        return('SeismologyButler(ID: {})'.format(self.periodogram.targetid))
+
+    @staticmethod
+    def from_lightcurve(lightcurve, **kwargs):
+        warnings.warn("Building a SeismologyButler object directly from a light curve "
+                      "using default periodogram parameters. For further tuneability, "
+                      "create a periodogram object first, using `to_periodogram`.")
+        return SeismologyButler(periodogram=lightcurve.to_periodogram())
+
+    @staticmethod
+    def from_periodogram(periodogram, **kwargs):
+        return SeismologyButler(periodogram=periodogram, **kwargs)
```

Comment:
I don't think there is a need for `from_periodogram` -- users should call the butler constructor directory!